<a href="https://colab.research.google.com/github/MCecileD/Food_NLP/blob/main/Photos_contr%C3%B4le.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import des bibliothèques nécessaires

In [2]:
import pandas as pd
import os
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Lister les fichiers/photos du répertoire

In [3]:
dir_photos = "drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/Food_Images/Food_Images"

In [5]:
!ls $dir_photos -1 | wc -l


13612


Création d'une boucle pour vérifier le nom des fichiers présents dans le répertoire

In [6]:
list_photos =[]
for file in os.listdir(f"{dir_photos}"):
  list_photos.append(file)

In [7]:
list_photos

['toasted-coconut-caramel-ice-cream-sundaes-353382.jpg',
 'toast-with-lemony-pea-mash-51154930.jpg',
 'toasted-manioc-flour-with-eggs-and-scallions-359470.jpg',
 'toasted-barley-risotto-with-spinach-and-herb-puree-51231690.jpg',
 'tiramisu-affogato-243203.jpg',
 'to-form-water-caltrop-wontons-51147800.jpg',
 'toasted-breadcrumbs-51140200.jpg',
 'tiramisu-with-amaretti-cookies.jpg',
 'tipsy-turtle-bark-233205.jpg',
 'toasted-flour-sables.jpg',
 'tiramisu-351138.jpg',
 'tin-roof-ice-cream-366134.jpg',
 'toasted-sweet-corn-pudding-356014.jpg',
 'toasted-chile-de-arbol-and-tomatillo-salsa.jpg',
 'toasted-spelt-soup-with-escarole-and-white-beans-51210020.jpg',
 'tofu-triangles-in-creamy-nut-buttter-sauce-with-scallions-355894.jpg',
 'timmys-brandy-milk-punch-356834.jpg',
 'tofu-stir-fry-351409.jpg',
 'toasted-coconut-chia-pudding.jpg',
 'toasted-coconut-cold-brew-iced-coffee.jpg',
 'toasted-almond-cake-with-strawberries-in-rose-water-syrup-238684.jpg',
 'toffee-squares-362389.jpg',
 'toaste

Création dataframe avec les noms des photos

In [8]:
df =  pd.DataFrame(columns=['Nom_entier','Nom_sans_format'])

In [9]:
df

Empty DataFrame
Columns: [Nom_entier, Nom_sans_format]
Index: []

In [10]:
df['Nom_entier'] = list_photos 

Formatter le nom des fichiers pour la comparaison

In [11]:
df['Nom_sans_format'] = df['Nom_entier'].str.slice(0,-4)

In [12]:
df

Nom_entier  \
0      toasted-coconut-caramel-ice-cream-sundaes-3533...   
1                toast-with-lemony-pea-mash-51154930.jpg   
2      toasted-manioc-flour-with-eggs-and-scallions-3...   
3      toasted-barley-risotto-with-spinach-and-herb-p...   
4                           tiramisu-affogato-243203.jpg   
...                                                  ...   
13607  baguette-smoked-oyster-and-pancetta-stuffing-5...   
13608                         bacon-baklava-51225820.jpg   
13609                      bacon-deviled-eggs-388689.jpg   
13610  bacon-wrapped-salmon-with-wilted-spinach-23380...   
13611               baked-alaska-saint-pierre-109544.jpg   

                                         Nom_sans_format  
0       toasted-coconut-caramel-ice-cream-sundaes-353382  
1                    toast-with-lemony-pea-mash-51154930  
2      toasted-manioc-flour-with-eggs-and-scallions-3...  
3      toasted-barley-risotto-with-spinach-and-herb-p...  
4                               tiramisu-affogato-243203  
...                                                  ...  
13607  baguette-smoked-oyster-and-pancetta-stuffing-5...  
13608                             bacon-baklava-51225820  
13609                          bacon-deviled-eggs-388689  
13610    bacon-wrapped-salmon-with-wilted-spinach-233802  
13611                   baked-alaska-saint-pierre-109544  

[13612 rows x 2 columns]

Répertoire du CSV

In [13]:
dir_csv = "drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/"

récupération du Food.csv

In [14]:
df_csv = pd.read_csv(dir_csv + '/Food.csv', delimiter=",")

In [15]:
df_csv

Unnamed: 0                                              Title  \
0               0  Miso-Butter Roast Chicken With Acorn Squash Pa...   
1               1                    Crispy Salt and Pepper Potatoes   
2               2                        Thanksgiving Mac and Cheese   
3               3                 Italian Sausage and Bread Stuffing   
4               4                                       Newton's Law   
...           ...                                                ...   
13496       13496                               Brownie Pudding Cake   
13497       13497  Israeli Couscous with Roasted Butternut Squash...   
13498       13498  Rice with Soy-Glazed Bonito Flakes and Sesame ...   
13499       13499                                        Spanakopita   
13500       13500  Mexican Poblano, Spinach, and Black Bean "Lasa...   

                                             Ingredients  \
0      ['1 (3½–4-lb.) whole chicken', '2¾ tsp. kosher...   
1      ['2 large egg whites', '1 pound new potatoes (...   
2      ['1 cup evaporated milk', '1 cup whole milk', ...   
3      ['1 (¾- to 1-pound) round Italian loaf, cut in...   
4      ['1 teaspoon dark brown sugar', '1 teaspoon ho...   
...                                                  ...   
13496  ['1 cup all-purpose flour', '2/3 cup unsweeten...   
13497  ['1 preserved lemon', '1 1/2 pound butternut s...   
13498  ['Leftover katsuo bushi (dried bonito flakes) ...   
13499  ['1 stick (1/2 cup) plus 1 tablespoon unsalted...   
13500  ['12 medium to large fresh poblano chiles (2 1...   

                                            Instructions  \
0      Pat chicken dry with paper towels, season all ...   
1      Preheat oven to 400°F and line a rimmed baking...   
2      Place a rack in middle of oven; preheat to 400...   
3      Preheat oven to 350°F with rack in middle. Gen...   
4      Stir together brown sugar and hot water in a c...   
...                                                  ...   
13496  Preheat the oven to 350°F. Into a bowl sift to...   
13497  Preheat oven to 475°F.\nHalve lemons and scoop...   
13498  If using katsuo bushi flakes from package, moi...   
13499  Melt 1 tablespoon butter in a 12-inch heavy sk...   
13500  Lay 4 chiles on their sides on racks of gas bu...   

                                              Image_Name  \
0      miso-butter-roast-chicken-acorn-squash-panzanella   
1             crispy-salt-and-pepper-potatoes-dan-kluger   
2             thanksgiving-mac-and-cheese-erick-williams   
3              italian-sausage-and-bread-stuffing-240559   
4                     newtons-law-apple-bourbon-cocktail   
...                                                  ...   
13496                         brownie-pudding-cake-14408   
13497  israeli-couscous-with-roasted-butternut-squash...   
13498  rice-with-soy-glazed-bonito-flakes-and-sesame-...   
13499                                 spanakopita-107344   
13500  mexican-poblano-spinach-and-black-bean-lasagne...   

                                     Cleaned_Ingredients  
0      ['1 (3½–4-lb.) whole chicken', '2¾ tsp. kosher...  
1      ['2 large egg whites', '1 pound new potatoes (...  
2      ['1 cup evaporated milk', '1 cup whole milk', ...  
3      ['1 (¾- to 1-pound) round Italian loaf, cut in...  
4      ['1 teaspoon dark brown sugar', '1 teaspoon ho...  
...                                                  ...  
13496  ['1 cup all-purpose flour', '2/3 cup unsweeten...  
13497  ['1 preserved lemon', '1 1/2 pound butternut s...  
13498  ['Leftover katsuo bushi (dried bonito flakes) ...  
13499  ['1 stick (1/2 cup) plus 1 tablespoon unsalted...  
13500  ['12 medium to large fresh poblano chiles (2 1...  

[13501 rows x 6 columns]

Création d'une colonne de présence des photos dans le CSV, on devrait obtenir 13501 images à True

In [16]:
df["PrésentdansCSV"] = df["Nom_sans_format"].isin(df_csv["Image_Name"])


Compter le nbre de True

In [17]:
len(df[df["PrésentdansCSV"]==True])

13471

In [18]:
len(df[df["PrésentdansCSV"]==False])

141

**On souhaite importer toutes les photos sans les filtrer dans le bucket de GCP et nous ferons un filtre en affichage sur le code de la page résultat du site web. Pour l'import : `CF Collab "import drive to bucket"`**





Vérification du CSV avec le répertoire, les Image_Name peuvent ne pas avoir de photos (contrôle caduque si on a le même nombre total de photos et recettes)

In [19]:
df_csv["Présentdansrepphotos"] = df_csv["Image_Name"].isin(df["Nom_sans_format"])

Compter le nbre de True => Image_Name dans le répertoire ?

In [20]:
len(df_csv[df_csv["Présentdansrepphotos"]==True])

13471

In [21]:
len(df_csv[df_csv["Présentdansrepphotos"]==False])

30